core/templates/dev/head/base_components/WarningLoader.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for warning_loader
+ */
+
+require('domain/utilities/UrlInterpolationService.ts');
+require('./PageFooter.ts');
+
+oppia.directive('warningLoader', [
+  'UrlInterpolationService', function(UrlInterpolationService) {
+  return {
+    restrict: 'E',
+    transclude: {
+      'content': 'content',
+      'footer': '?pageFooter'

```

Comment:
Why warning content contains content and footer? I think that content and footer should be transclusive part of base content and warning should be separate directive (like footer) that's inside base content but not transclusive.

Reply:
Okay.
I made it that way since that is how it is done in the macro in base.html

Reply:
Also, the content and footer has to be a child element here https://github.com/oppia/oppia/blob/18f8fb33b64237c79e59e291874e16db4f132079/core/templates/dev/head/pages/base.html#L2

Reply:
What do you mean by this?

Reply:
Currently, the warning_loader macro includes the blocks for content and footer
So I decided to make this one that way

Reply:
Okay, it is quite an antipattern, please rework so that content and footer are outside the warning

Reply:
Okay